File: Assets/common/script/ship/JammingSystem.js
#pragma strict
interface JammingListener {

	 function jammingResult(res :int) ;
}

class JammingSystem extends BaseSubsystem implements JammingListener{
	
	private var completeCallback : JammingListener = this;
	var running : boolean = false;
	
	function Start () {
		super.Start();
	}
	
	function Update () {
	
	}
	
	function startJammer(cb : JammingListener){
		completeCallback = cb;
		
		OSCHandler.Instance.ChangeClientScreen("EngineerStation", "jamming");
		running = true;
	}
	
	function gotResult(res : int){
		Debug.Log("got result" + res);
		if(running){
			completeCallback.jammingResult(res);
			Debug.Log("Done cb");
			completeCallback = this;
			running = false;
		}
	}
	
	function jammingResult(success : int){
		Debug.Log("scanner finished with : " + success);
	}
	
	function processOSCMessage(message : OSCMessage){
		var msgAddress = message.Address.Split(separator);
		// [1] = System, 2 = Subsystem name, 3 = operation
		var system = msgAddress[2];
		var target = msgAddress[3];
		switch(target){
			case "jamresult":
				var r : int = message.Data[0];
				gotResult(r);
				break;
		}
		
	}
	
}

File: Assets/common/script/ship/JammingSystem.cs
using UnityEngine;
using System;
using UnityOSC;


[System.Serializable]
public class JammingSystem: BaseSubsystem , JammingListener{
	
	JammingListener completeCallback;
	public bool running = false;
	
	public override void Start() {
		base.Start();
			completeCallback = this;
	}
	
	public void Update() {
	
	}
	
	public void startJammer(JammingListener cb,int diff){
		completeCallback = cb;
		
		OSCHandler.Instance.ChangeClientScreen("EngineerStation", "jamming", true);
		OSCMessage o = new OSCMessage("/system/jammer/setDifficulty");
		o.Append<int>(diff);
		OSCHandler.Instance.SendMessageToAll(o);
		running = true;
	}
	
	public void gotResult(int res){
		UnityEngine.Debug.Log("got result" + res);
		if(running){
			OSCHandler.Instance.RevertClientScreen("EngineerStation", "jamming");
			completeCallback.jammingResult(res);
			UnityEngine.Debug.Log("Done cb");
			completeCallback = this;
			running = false;
		}
	}
	
	public void jammingResult(int success){
		UnityEngine.Debug.Log("scanner finished with : " + success);
	}
	
	public override void processOSCMessage(OSCMessage message){
		string[] msgAddress = message.Address.Split(separator);
		// [1] = System, 2 = Subsystem name, 3 = operation
		string system = msgAddress[2];
		string target = msgAddress[3];
		switch(target){
			case "jamresult":
				int r = (int)message.Data[0];
				gotResult(r);
				break;
		}
		
	}
	
}
